src/components/NetworkDisplay.tsx

```diff
@@ -1,36 +1,42 @@
+import { Button, Typography, useTheme } from "@mui/material"
 import { ChainId, IS_L2_SUPPORTED } from "../constants"
-import React, { ReactElement } from "react"
 import { UnsupportedChainIdError, useWeb3React } from "@web3-react/core"
 
-import Button from "./Button"
+import CircleTwoToneIcon from "@mui/icons-material/CircleTwoTone"
 import { NETWORK_LABEL } from "../constants/networks"
-import classnames from "classnames"
-import styles from "./NetworkDisplay.module.scss"
+import React from "react"
 
-export default function NetworkDisplay({
-  onClick,
-}: {
-  onClick: () => void
-}): ReactElement | null {
-  const { active, chainId, error } = useWeb3React()
-  const networkLabel: string =
-    (chainId ? NETWORK_LABEL[chainId as ChainId] : undefined) ?? "Ethereum"
-  const isUnsupportChainIdError = error instanceof UnsupportedChainIdError
-
-  return IS_L2_SUPPORTED ? (
-    <Button
-      data-testid="networkDisplayBtn"
-      kind="ghost"
-      size="medium"
-      onClick={onClick}
-    >
-      <div
-        className={classnames(styles.circle, {
-          [styles.wrong]: isUnsupportChainIdError,
-          [styles.active]: active,
-        })}
-      ></div>{" "}
-      {networkLabel}
-    </Button>
-  ) : null
+interface NetworkDisplayProps extends React.ComponentPropsWithoutRef<"button"> {
+  onClick: React.MouseEventHandler<HTMLButtonElement>
 }
+const NetworkDisplay = React.forwardRef<HTMLButtonElement, NetworkDisplayProps>(
+  function NetworkDisplay({ onClick }, ref) {
+    const { active, chainId, error } = useWeb3React()
+    const theme = useTheme()
+    const networkLabel: string =
+      (chainId ? NETWORK_LABEL[chainId as ChainId] : undefined) ?? "Ethereum"
+    const isUnsupportChainIdError = error instanceof UnsupportedChainIdError
+
+    return IS_L2_SUPPORTED ? (
```

Comment:
you can also do 

`IS_L2_SUPPORTED && (...)` here to slightly shorten syntax. `? (...) : (...)` is good for switching components around.

Reply:
Updated!